test/TestAzimuth.js

```diff
@@ -182,30 +182,47 @@ contract('Azimuth', function([owner, user, user2, user3]) {
   });
 
   it('setting keys', async function() {
-    let [crypt, auth, suite, rev] = await azimuth.getKeys(0);
+    let { crypt, auth, suite, revision } = await azimuth.getKeys(0);
     assert.equal(crypt,
       '0x0000000000000000000000000000000000000000000000000000000000000000');
     assert.equal(auth,
       '0x0000000000000000000000000000000000000000000000000000000000000000');
     assert.equal(suite, 0);
-    assert.equal(rev, 0);
+    assert.equal(revision, 0);
     assert.equal(await azimuth.getKeyRevisionNumber(0), 0);
     assert.isFalse(await azimuth.isLive(0));
     // only owner can do this.
-    await assertRevert(azimuth.setKeys(0, 10, 11, 2, {from:user}));
-    await seeEvents(azimuth.setKeys(0, 10, 11, 2), ['ChangedKeys']);
-    await seeEvents(azimuth.setKeys(0, 10, 11, 2), []);
-    [crypt, auth, suite, rev] = await azimuth.getKeys(0);
+    await assertRevert(azimuth.setKeys(web3.utils.toHex(0),
+                                       web3.utils.toHex(10),
+                                       web3.utils.toHex(11),
+                                       web3.utils.toHex(2),
+                                       {from:user}));
+    await seeEvents(azimuth.setKeys(web3.utils.toHex(0),
+                                    web3.utils.toHex(10),
+                                    web3.utils.toHex(11),
+                                    web3.utils.toHex(2)), ['ChangedKeys']);
+    await seeEvents(azimuth.setKeys(web3.utils.toHex(0),
+                                    web3.utils.toHex(10),
+                                    web3.utils.toHex(11),
+                                    web3.utils.toHex(2)), []);
+    let ks = await azimuth.getKeys(web3.utils.toHex(0));
+    crypt = ks.crypt;
+    auth = ks.auth;
+    suite = ks.suite;
+    revision = ks.revision;
     assert.equal(crypt,
-      '0xa000000000000000000000000000000000000000000000000000000000000000');
+      '0x0a00000000000000000000000000000000000000000000000000000000000000');
```

Comment:
These right shifts are weird, happen in three places and no idea what causes this! Everything else is just boring breaking api changes.

Reply:
Looks like it's right-padding on the byte rather than half-byte or string level now?